orienteer-core/src/main/java/org/orienteer/core/service/OModuleExecutorInitModule.java

```diff
@@ -0,0 +1,180 @@
+package org.orienteer.core.service;
+
+import com.google.common.collect.Lists;
+import com.google.inject.*;
+import com.google.inject.name.Named;
+import com.google.inject.name.Names;
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory;
+import org.eclipse.aether.spi.connector.transport.TransporterFactory;
+import org.eclipse.aether.transport.file.FileTransporterFactory;
+import org.eclipse.aether.transport.http.HttpTransporterFactory;
+import org.orienteer.core.loader.Dependency;
+import org.orienteer.core.loader.MavenResolver;
+import org.orienteer.core.loader.OrienteerOutsideModulesManager;
+import org.orienteer.core.loader.OrienteerOutsideModules;
+import org.orienteer.core.loader.util.ConsoleRepositoryListener;
+import org.orienteer.core.loader.util.ConsoleTransferListener;
+import org.orienteer.core.loader.util.PomXmlParser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xeustechnologies.jcl.proxy.CglibProxyProvider;
+import org.xeustechnologies.jcl.proxy.ProxyProviderFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+
+/**
+ * @author Vitaliy Gonchar
+ */
+public class OModuleExecutorInitModule extends AbstractModule {
+
+
+    private final static Logger LOG = LoggerFactory.getLogger(OModuleExecutorInitModule.class);
+
+    private final String localRepositoryPath = System.getProperty("user.home") + "/.m2/repository/";
+//    private final String localRepositoryPath = System.getProperty("user.dir") + "/tmp/dependencies/";
+    private final String parentOrienteerPom = "../pom.xml";
+
+    private final Properties properties;
+
+    public OModuleExecutorInitModule(Properties properties) {
+        this.properties = properties;
+    }
+
+    public OModuleExecutorInitModule() {
+        this.properties = null;
+    }
+
+    @Override
+    protected void configure() {
+        bind(RepositorySystem.class).toProvider(RepositorySystemProvider.class).in(Singleton.class);
```

Comment:
I see that properties file is being loaded in child Injector - not here. Why you selected this way? Probably properties file can be used for configuring some of the mentioned things here?